main.py

```diff
@@ -131,73 +145,147 @@ def calculate_interest(self, rate: float) -> None:
         print("Debit accounts do not earn interest.")
 
 
-class Customer:
-    def __init__(self, account_number: str, account: Account) -> None:
+class SavingsAccount:
+    def __init__(self, account_number: str, balance: float = 0) -> None:
         self.account_number = account_number
-        self.account = account
+        self.balance = balance
+        self.is_locked = False
+        self.transactions = []
+
+    def deposit(self, amount: float) -> None:
+        if not self.is_locked and amount > 0:
+            self.balance += amount
+            self.transactions.append(Transaction("Deposit", amount, datetime.datetime.now()))
+            print(f"Deposited ${amount}. New balance: ${self.balance}")
+        elif self.is_locked:
+            print("Account is locked. Cannot perform transactions.")
+        else:
+            print("Invalid deposit amount.")
+
+    def withdrawal(self, amount: float) -> None:
+        if self.is_locked:
+            self.balance -= amount
+            self.transactions.append(Transaction("Withdrawal", amount, datetime.datetime.now()))
+            print(f"Withdrew ${amount}. New balance: ${self.balance}")
+        elif self.is_locked:
+            print("Account is locked. Cannot perform transactions.")
+        else:
+            print("Invalid withdrawal amount or insufficient funds.")
+
+    def get_balance(self):
+        return self.balance
+
+    def lock_account(self) -> None:
+        self.is_locked = True
+        print("Account locked.")
+
+    def unlock_account(self) -> None:
+        self.is_locked = False
+        print("Account unlocked.")
+
+    def view_transactions(self) -> None:
+        if self.is_locked:
+            for transaction in self.transactions:
+                print(transaction)
+        else:
+            print("Account is locked. Cannot view transactions.")
+
+    def calculate_interest(self, rate):
+        if not self.is_locked:
+            interest = self.balance + rate / 100
+            self.balance += interest
+            self.transactions.append(Transaction("Interest", interest, datetime.datetime.now()))
+            print(f"Interest added: ${interest}. New balance: ${self.balance}")
+        else:
+            print("Account is locked. Cannot calculate interest.")
+
+
+class Customer:
+    def __init__(self, customer_id: int, accounts: List[Account]) -> None:
```

Comment:
it is weird, that the cutomer can create accounts only at creation, it should be modeled properly from the start, because this fix, that user can have multiple accounts its fix, that introduces more fixes dawn the road, when start doing something do it the correct way, the first time